src/pal/src/misc/sysinfo.cpp

```diff
@@ -403,5 +450,47 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     cacheSize = max(cacheSize, sysconf(_SC_LEVEL4_CACHE_SIZE));
 #endif
 
+    if(cacheSize == 0)
+    {
+        size_t size;
+
+        if(PAL_ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index0/size", &size))
+            cacheSize = max(cacheSize, size);
+        if(PAL_ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index1/size", &size))
+            cacheSize = max(cacheSize, size);
+        if(PAL_ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index2/size", &size))
+            cacheSize = max(cacheSize, size);
+        if(PAL_ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index3/size", &size))
+            cacheSize = max(cacheSize, size);
+        if(PAL_ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index4/size", &size))
+            cacheSize = max(cacheSize, size);
```

Comment:
@janvorli Hardcoded paths, philisophically sysfs might be mounted on a different mount point, but this seems to be the cannonical path.

Some of these index correspond to ICache, but they would be small caches anyway, so I ignored that.

index4 should be L4$ which seems big enough for current processors.